File: app/src/main/java/com/kirchhoff/animation/scaling/ScalingMainActivity.java
package com.kirchhoff.animation.scaling;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.kirchhoff.animation.R;

/**
 * @author Kirchhoff-
 */

public class ScalingMainActivity extends AppCompatActivity {

    public static void start(Context context) {
        Intent intent = new Intent(context, ScalingMainActivity.class);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.a_scaling_main);
    }

    public void scalingLayoutFab(View view) {
        ScalingFab.Companion.start(this);
    }

    public void scalingLayoutSearchBar(View view) {
        ScalingSearchBar.start(this);
    }

    public void scalingLayoutCollapsing(View view) {
        ScalingCollapsing.start(this);
    }
}

File: scalinglayout/src/main/java/com/kirchhoff/scalinglayout/ScalingMainActivity.kt
package com.kirchhoff.scalinglayout

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View

class ScalingMainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.a_scaling_main)
        findViewById<View>(R.id.bScalingLayoutFab).setOnClickListener(this)
        findViewById<View>(R.id.bScalingLayoutSearchBar).setOnClickListener(this)
        findViewById<View>(R.id.bScalingLayoutCollapsing).setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.bScalingLayoutFab -> startActivity(ScalingFab::class.java)
            R.id.bScalingLayoutSearchBar -> startActivity(ScalingSearchBar::class.java)
            R.id.bScalingLayoutCollapsing -> startActivity(ScalingCollapsing::class.java)
        }
    }

    private fun startActivity(activityClass: Class<*>) {
        val intent = Intent(this, activityClass)
        startActivity(intent)
    }
}
